### PR TITLE
Add information received in response to prepare to the trace

### PR DESCRIPTION
**Pull request by smaczliwka**
The PR introduces parsing operation type received in custom payload in response to prepare and setting it as a tag of bound statement. The incoming payload of prepared statement is propagated as outgoing payload if no outgoing payload has been explicitly set but then operation type is removed from the payload that is propagated.

### CODE REVIEW

File: driver-core/src/main/java/com/datastax/driver/core/BoundStatement.java [hunk 1]
```diff
@@ -92,18 +95,32 @@ public BoundStatement(PreparedStatement statement) {
...
     if (statement.getRetryPolicy() != null) this.setRetryPolicy(statement.getRetryPolicy());
+
+    Map<String, ByteBuffer> incomingPayload =
+        (statement.getIncomingPayload() != null)
+            ? new HashMap<String, ByteBuffer>(statement.getIncomingPayload())
+            : null;
+    if (incomingPayload != null && incomingPayload.containsKey("db.operation")) {
+      this.operationType = new String(incomingPayload.remove("db.operation").array());
+      if (incomingPayload.isEmpty()) incomingPayload = null;
+    } else this.operationType = null;
```
Comment (constantfold) [inline driver-core/src/main/java/com/datastax/driver/core/BoundStatement.java]: I'd move this logic to `PreparedStatement` for two reasons:
1. `PreparedStatement` received this info and is an "owner" of it, not `BoundStatement`
2. Performance - better to preprocess `incomingPayload` and parse it once (preparing a query is done once) vs. do this processing for each `BoundStatement` (thousands/millions of invocations)

**Review by avelanarius — APPROVED** [review]

**Comment by avelanarius** [comment]
@smaczliwka Please rebase on top of latest master.

**Comment by avelanarius** [comment]
and also modify the parsing code to work correctly with https://github.com/zpp-2021-telemetry/scylla/pull/10

**Comment by avelanarius** [comment]
Now, when testing with https://github.com/zpp-2021-telemetry/scylla/pull/10, there is one problem driver-side: `opentelemetry` custom payload now contains both: list of replicas, operation type. However, the parsing code doesn't correctly parse those values separately - the operation type string contains the bytes of the list of replicas.

The way I reproduced it: 

2 Scylla nodes started (code from https://github.com/zpp-2021-telemetry/scylla/pull/10). 
```
CREATE KEYSPACE ks2 WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': 2};
CREATE TABLE ks2.t(pk int, ck int, v int, PRIMARY KEY(pk, ck));
```
```
        PreparedStatement ps = s.prepare("INSERT INTO ks2.t(pk, ck, v) VALUES (?, ?, ?)");
        System.out.println("Operation type is: " + ps.getOperationType());
```
which now prints:
image
those first 4 characters are probably the bytes of replica list.

**Pull request merged**